Add "View Details" button for collections

mvdbeek opened the pull request:
Which loads up the JobDetails.vue component. This is super-useful for
dynamic collections that failed, where you had no chance to see the
stderr or stdout.
Currently only available in the new history panel.

![Screenshot 2021-07-15 at 21 39 40](https://user-images.githubusercontent.com/6804901/125849308-e15f883a-75ac-407c-a44a-7a69da1df012.png)


## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

mvdbeek committed:
Add "View Details" button for collections

Which loads up the JobDetails.vue component. This is super-useful for
dynamic collections that failed, where you had no chance to see the
stderr or stdout.

mvdbeek committed:
Move JobDetails out of admin component

jmchilton commented on client/src/components/History/ContentItem/DatasetCollection/DscMenu.vue:
```diff
@@ -1,18 +1,53 @@
 <template>
-    <b-dropdown no-caret right variant="link" size="sm" boundary="window" toggle-class="p-1">
```
Three open PRs from three authors with changes to make this a priority menu 😆. I think I've isolated the changes from Assunta in 8457c29f67e7bcf830bee353b2792a30b1c18376 if you can rebase on top of that? I meant to open a PR that was just this.

mvdbeek replied:
I think the PriorityMenu was dropped ultimately in the collapsed view, so I guess this is all moot at this point.

mvdbeek committed:
Better URL construction

mvdbeek committed:
Show outputs as well

mvdbeek committed:
Display only non-hidden outputs, display input datasets

mvdbeek committed:
This seems to fix all the routing issues?

mvdbeek committed:
Replace parameter list with table

mvdbeek committed:
Adjust client unit tests

mvdbeek committed:
Add JobOutputs

mvdbeek committed:
Not using this anymore

mvdbeek committed:
Use div instead of tested tr/td

mvdbeek committed:
Use click.stop.prevent to avoid nested iframe

mvdbeek committed:
Load center panel instead of trying to load datasets/edit if no dataset id in query params

mvdbeek committed:
Display job outputs in-line with JobParameters

I think this makes sense, and we can label the output with the `label`
if it is available for an output.

mvdbeek committed:
Sanitize client structure

assuntad23 commented on client/src/components/History/ContentItem/DatasetCollection/DscMenu.vue:
```diff
@@ -1,18 +1,51 @@
-            <span class="sr-only">Delete Collection</span>
-        </template>
+    <PriorityMenu>
+        <b-dropdown no-caret right variant="link" size="sm" boundary="window" toggle-class="p-1">
```
@mvdbeek  Does the b-dropdown work inside of the PriorityMenu? I had trouble with it, where it would drop down a menu, but the menu would be empty  So, I moved all of the delete functions into a modal that popped up. If this works though, I'll get rid of that.

mvdbeek replied:
It did, but I think we dropped use of the PriorityMenu in the collapsed view.

mvdbeek committed:
Fix imports

mvdbeek committed:
Fix spacing of buttons

mvdbeek committed:
Drop unnecessary blank lines

mvdbeek committed:
Right-align b-dropdown for dataset operations

Otherwhise the menu may flow out of the center iframe in the job outputs
view.

mvdbeek committed:
Test jobOutputs component

mvdbeek committed:
Selenium test for collection details view

mvdbeek commented on client/src/app/singleton.js:
```diff
@@ -39,5 +39,5 @@ export function galaxyIsInitialized() {
 // of galaxy in each window for the short-term
 export function getStorage() {
-    return window;
+    return window.parent;
```
@dannon this was necessary so that clicking on the rerun button from within the center iframe won't nest the Galaxy app twice (2 mastheads, 2 tool panel etc) ... it seems this doesn't break anything else, but I also have no clue whether that's correct.

dannon replied:
I don't remember all the details right now, but I think this *probably* breaks what this was intended to fix by always returning a single toplevel Galaxy object.

dannon replied:
The source PR was this:  https://github.com/galaxyproject/galaxy/pull/7135, but I think that predates the shift to the new tool install interface, so we might be okay to actually revert all that and clean up/simplify a bit.
